app/models/ability.rb

```diff
@@ -0,0 +1,3 @@
+class Ability
+  include CanCan::Ability
+end
```

Comment:
- [ ] Add initialize method in ability class that will consist of all rules for the authorization. You can consider the below code.
```ruby
def initialize(user)

         user ||= User.new # guest user (not logged in)

         if user.admin? 
         can :manage, :all
         else
         can :manage, Post, author: user  # owner of post can perform all view, update or destroy own post if Post.author == user
         can :manage, Comment, author: user
         can :read, :all
         end
     end